--- BlockWorkbench.java
+++ BlockWorkbench.java
@@ -1,3 +1,3 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 public class BlockWorkbench extends Block {
@@ -6,10 +6,10 @@
         this.blockIndexInTexture = 59;
     }
-    
+
     @Override
     public int getBlockTextureFromSide(final int side) {
-        return (side == 1) ? (this.blockIndexInTexture - 16) : ((side == 0) ? Block.planks.getBlockTextureFromSide(0) : ((side != 2 && side != 4) ? this.blockIndexInTexture : (this.blockIndexInTexture + 1)));
+        return (side == 1) ? (this.blockIndexInTexture - 16) : ((side == 0) ? Block.PLANKS.getBlockTextureFromSide(0) : ((side != 2 && side != 4) ? this.blockIndexInTexture : (this.blockIndexInTexture + 1)));
     }
-    
+
     @Override
     public boolean blockActivated(final World world, final int x, final int y, final int z, final EntityPlayer entityPlayer) {
