--- ConsoleLogFormatter.java
+++ ConsoleLogFormatter.java
@@ -1,3 +1,3 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 import java.io.Writer;
@@ -11,9 +11,9 @@
 final class ConsoleLogFormatter extends Formatter {
     private SimpleDateFormat dateFormat;
-    
+
     ConsoleLogFormatter() {
         this.dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
     }
-    
+
     public String format(final LogRecord logRecord) {
         final StringBuilder stringBuilder2 = new StringBuilder();
@@ -39,5 +39,5 @@
         }
         else if (level3 == Level.SEVERE) {
-            stringBuilder2.append(new StringBuilder(" [").append(level3.getLocalizedName()).append("] ").toString());
+            stringBuilder2.append(" [" + level3.getLocalizedName() + "] ");
         }
         stringBuilder2.append(logRecord.getMessage());
@@ -46,6 +46,6 @@
         if (throwable4 != null) {
             final StringWriter stringWriter5 = new StringWriter();
-            throwable4.printStackTrace(new PrintWriter((Writer)stringWriter5));
-            stringBuilder2.append(stringWriter5.toString());
+            throwable4.printStackTrace(new PrintWriter(stringWriter5));
+            stringBuilder2.append(stringWriter5);
         }
         return stringBuilder2.toString();
