--- EntityCreeper.java
+++ EntityCreeper.java
@@ -1,3 +1,3 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 public class EntityCreeper extends EntityMob {
@@ -6,5 +6,5 @@
     int fuseDuration;
     int creeperState;
-    
+
     public EntityCreeper(final World world) {
         super(world);
@@ -13,15 +13,15 @@
         this.texture = "/mob/creeper.png";
     }
-    
+
     @Override
     public void writeEntityToNBT(final NBTTagCompound nbttagcompound) {
         super.writeEntityToNBT(nbttagcompound);
     }
-    
+
     @Override
     public void readEntityFromNBT(final NBTTagCompound nbttagcompound) {
         super.readEntityFromNBT(nbttagcompound);
     }
-    
+
     @Override
     protected void updateEntityActionState() {
@@ -38,15 +38,15 @@
         }
     }
-    
+
     @Override
     protected String getHurtSound() {
         return "mob.creeper";
     }
-    
+
     @Override
     protected String getDeathSound() {
         return "mob.creeperdeath";
     }
-    
+
     @Override
     public void onDeath(final Entity entity) {
@@ -56,5 +56,5 @@
         }
     }
-    
+
     @Override
     protected void attackEntity(final Entity entity, final float damage) {
@@ -72,5 +72,5 @@
         }
     }
-    
+
     @Override
     protected int getDropItemId() {
