--- ItemBlock.java
+++ ItemBlock.java
@@ -1,7 +1,7 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 public class ItemBlock extends Item {
     private int blockID;
-    
+
     public ItemBlock(final int itemID) {
         super(itemID);
@@ -9,8 +9,8 @@
         this.setIconIndex(Block.blocksList[itemID + 256].getBlockTextureFromSide(2));
     }
-    
+
     @Override
     public boolean onItemUse(final ItemStack stack, final EntityPlayer entityPlayer, final World world, int x, int y, int z, int i7) {
-        if (world.getBlockId(x, y, z) == Block.snow.blockID) {
+        if (world.getBlockId(x, y, z) == Block.SNOW.blockID) {
             i7 = 0;
         }
