--- ItemReed.java
+++ ItemReed.java
@@ -1,15 +1,15 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 public class ItemReed extends Item {
     private int spawnID;
-    
+
     public ItemReed(final int itemID, final Block reed) {
         super(itemID);
         this.spawnID = reed.blockID;
     }
-    
+
     @Override
     public boolean onItemUse(final ItemStack stack, final EntityPlayer entityPlayer, final World world, int x, int y, int z, int i7) {
-        if (world.getBlockId(x, y, z) == Block.snow.blockID) {
+        if (world.getBlockId(x, y, z) == Block.SNOW.blockID) {
             i7 = 0;
         }
