--- ItemDoor.java
+++ ItemDoor.java
@@ -1,7 +1,7 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 public class ItemDoor extends Item {
     private Material material;
-    
+
     public ItemDoor(final int itemID, final Material material) {
         super(itemID);
@@ -10,5 +10,5 @@
         this.maxStackSize = 1;
     }
-    
+
     @Override
     public boolean onItemUse(final ItemStack stack, final EntityPlayer entityPlayer, final World world, final int x, int y, final int z, final int i7) {
@@ -19,8 +19,8 @@
         Block block8;
         if (this.material == Material.wood) {
-            block8 = Block.doorWood;
+            block8 = Block.DOOR_WOOD;
         }
         else {
-            block8 = Block.doorSteel;
+            block8 = Block.DOOR_IRON;
         }
         if (!block8.canPlaceBlockAt(world, x, y, z)) {
@@ -42,6 +42,6 @@
             b10 = 1;
         }
-        final int i9 = (world.isBlockNormalCube(x - b10, y, z - b11) + world.isBlockNormalCube(x - b10, y + 1, z - b11)) ? 1 : 0;
-        final int i10 = (world.isBlockNormalCube(x + b10, y, z + b11) + world.isBlockNormalCube(x + b10, y + 1, z + b11)) ? 1 : 0;
+final int i9 = (world.isBlockNormalCube(x - b10, y, z - b11) && world.isBlockNormalCube(x - b10, y + 1, z - b11)) ? 1 : 0;
+final int i10 = (world.isBlockNormalCube(x + b10, y, z + b11) && world.isBlockNormalCube(x + b10, y + 1, z + b11)) ? 1 : 0;
         final boolean z2 = world.getBlockId(x - b10, y, z - b11) == block8.blockID || world.getBlockId(x - b10, y + 1, z - b11) == block8.blockID;
         final boolean z3 = world.getBlockId(x + b10, y, z + b11) == block8.blockID || world.getBlockId(x + b10, y + 1, z + b11) == block8.blockID;
