--- ItemMinecart.java
+++ ItemMinecart.java
@@ -1,7 +1,7 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 public class ItemMinecart extends Item {
     public int minecartType;
-    
+
     public ItemMinecart(final int itemID, final int minecartType) {
         super(itemID);
@@ -9,9 +9,9 @@
         this.minecartType = minecartType;
     }
-    
+
     @Override
     public boolean onItemUse(final ItemStack stack, final EntityPlayer entityPlayer, final World world, final int x, final int y, final int z, final int i7) {
         final int i8 = world.getBlockId(x, y, z);
-        if (i8 == Block.minecartTrack.blockID) {
+        if (i8 == Block.MINECART_TRACK.blockID) {
             world.spawnEntityInWorld(new EntityMinecart(world, x + 0.5f, y + 0.5f, z + 0.5f, this.minecartType));
             --stack.stackSize;
