--- MCHashEntry2.java
+++ MCHashEntry2.java
@@ -1,3 +1,3 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 class MCHashEntry2 {
@@ -6,5 +6,5 @@
     MCHashEntry2 nextEntry;
     final int slotHash;
-    
+
     MCHashEntry2(final int slotHash, final long hashEntry, final Object valueEntry, final MCHashEntry2 nextEntry) {
         this.valueEntry = valueEntry;
@@ -13,13 +13,13 @@
         this.slotHash = slotHash;
     }
-    
+
     public final long getHash() {
         return this.hashEntry;
     }
-    
+
     public final Object getValue() {
         return this.valueEntry;
     }
-    
+
     public final boolean equals(final Object object) {
         if (!(object instanceof MCHashEntry2)) {
@@ -38,9 +38,9 @@
         return false;
     }
-    
+
     public final int hashCode() {
         return MCHashTable2.getHash(this.hashEntry);
     }
-    
+
     public final String toString() {
         return String.valueOf(this.getHash()) + "=" + this.getValue();
