--- MCHashEntry.java
+++ MCHashEntry.java
@@ -1,3 +1,3 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 class MCHashEntry {
@@ -6,5 +6,5 @@
     MCHashEntry nextEntry;
     final int slotHash;
-    
+
     MCHashEntry(final int slotHash, final int hashEntry, final Object valueEntry, final MCHashEntry nextEntry) {
         this.valueEntry = valueEntry;
@@ -13,13 +13,13 @@
         this.slotHash = slotHash;
     }
-    
+
     public final int getHash() {
         return this.hashEntry;
     }
-    
+
     public final Object getValue() {
         return this.valueEntry;
     }
-    
+
     public final boolean equals(final Object object) {
         if (!(object instanceof MCHashEntry)) {
@@ -38,9 +38,9 @@
         return false;
     }
-    
+
     public final int hashCode() {
         return MCHashTable.getHash(this.hashEntry);
     }
-    
+
     public final String toString() {
         return String.valueOf(this.getHash()) + "=" + this.getValue();
