--- Namegen.java
+++ Namegen.java
@@ -1,5 +1,4 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
-import net.minecraft.server.MinecraftServer;
 import java.util.Random;
 
@@ -22,10 +21,10 @@
     public long areaTimer;
     private static final String[] syllab;
-    
+
     static {
         Namegen.uqKey = "";
         syllab = new String[] { "SIE", "LOH", "KII", "HUR", "MIS", "RUU", "VY", "KA", "TAV", "OLE", "PAH", "MUI", "MAT", "JA", "SAU", "NIN", "UD", "MU", "NGI", "BAR", "LUG", "MAH", "GIR", "AK", "USU", "ESE", "IRU", "UUN", "AMTU", "AGAS", "HI", "TOOI", "YORU", "NEN", "PON", "ONNA", "TSU", "YA", "AO", "ONI", "AN", "KO", "SHI", "YUME", "YARI", "TEST" };
     }
-    
+
     public Namegen() {
         this.rand = new Random();
@@ -40,5 +39,5 @@
         this.areaTimer = 0L;
     }
-    
+
     public static String Namegen2(final long l, final int n, final int n2) {
         final long l2 = n + 392214;
@@ -46,8 +45,8 @@
         final long l4 = l3 * 784428L + l2;
         final long l5 = l + l4;
-        final int n3 = (int)Math.sqrt((double)(n * n + n2 * n2));
+        final int n3 = (int)Math.sqrt(n * n + n2 * n2);
         int n4 = 0;
         int n5 = 3;
-        for (int n6 = n3; n6 / n5 > 0; n5 *= 4) {
+        for (; n3 / n5 > 0; n5 *= 4) {
             ++n4;
         }
@@ -55,14 +54,14 @@
         if (n4 > 0) {
             for (int i = 0; i != n4; ++i) {
-                string = String.valueOf(string) + Namegen.syllab[(int)((n3 + n4 + n5 + l4 * (2 + i)) % 45L)];
+                string = string + Namegen.syllab[(int)((n3 + n4 + n5 + l4 * (2 + i)) % 45L)];
             }
-            string = String.valueOf(string) + "-";
+            string = string + "-";
         }
-        string = String.valueOf(string) + Namegen.syllab[(int)((l4 * 2L + n3) % 45L)];
-        string = String.valueOf(string) + Namegen.syllab[(int)((l4 * 3L + n3) % 45L)];
-        string = String.valueOf(string) + Namegen.syllab[(int)((l4 * 4L + n3) % 45L)];
+        string = string + Namegen.syllab[(int)((l4 * 2L + n3) % 45L)];
+        string = string + Namegen.syllab[(int)((l4 * 3L + n3) % 45L)];
+        string = string + Namegen.syllab[(int)((l4 * 4L + n3) % 45L)];
         return string;
     }
-    
+
     public void ResetNamegen() {
         final float playerPosX = 0.0f;
@@ -73,8 +72,8 @@
         this.initedArea = true;
         this.currentArea = Namegen2(this.lastSeed, this.lgroupX, this.lgroupY);
-        System.out.println(new StringBuilder("Name at ").append(playerPosX).append(" ").append(playerPosZ).append(": ").append(this.currentArea).toString());
+        System.out.println("Name at " + playerPosX + " " + playerPosZ + ": " + this.currentArea);
         this.areaTimer = System.currentTimeMillis();
     }
-    
+
     public boolean NamegenNeedsReset() {
         final float playerPosX = 0.0f;
