--- Packet33RelEntityMoveLook.java
+++ Packet33RelEntityMoveLook.java
@@ -1,3 +1,3 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 import java.io.DataOutputStream;
@@ -9,5 +9,5 @@
         this.rotating = true;
     }
-    
+
     public Packet33RelEntityMoveLook(final int entityID, final byte xPosition, final byte yPosition, final byte zPosition, final byte yaw, final byte pitch) {
         super(entityID);
@@ -19,5 +19,5 @@
         this.rotating = true;
     }
-    
+
     @Override
     public void readPacketData(final DataInputStream dataInputStream) throws IOException {
@@ -29,15 +29,15 @@
         this.pitch = dataInputStream.readByte();
     }
-    
+
     @Override
     public void writePacket(final DataOutputStream dataOutputStream) throws IOException {
         super.writePacket(dataOutputStream);
-        dataOutputStream.writeByte((int)this.xPosition);
-        dataOutputStream.writeByte((int)this.yPosition);
-        dataOutputStream.writeByte((int)this.zPosition);
-        dataOutputStream.writeByte((int)this.yaw);
-        dataOutputStream.writeByte((int)this.pitch);
+        dataOutputStream.writeByte(this.xPosition);
+        dataOutputStream.writeByte(this.yPosition);
+        dataOutputStream.writeByte(this.zPosition);
+        dataOutputStream.writeByte(this.yaw);
+        dataOutputStream.writeByte(this.pitch);
     }
-    
+
     @Override
     public int getPacketSize() {
