--- Packet5PlayerInventory.java
+++ Packet5PlayerInventory.java
@@ -1,3 +1,3 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 import java.io.DataOutputStream;
@@ -8,8 +8,8 @@
     public int inventoryType;
     public ItemStack[] inventory;
-    
+
     public Packet5PlayerInventory() {
     }
-    
+
     public Packet5PlayerInventory(final int inventoryType, final ItemStack[] stacks) {
         this.inventoryType = inventoryType;
@@ -19,5 +19,5 @@
         }
     }
-    
+
     @Override
     public void readPacketData(final DataInputStream dataInputStream) throws IOException {
@@ -34,26 +34,25 @@
         }
     }
-    
+
     @Override
     public void writePacket(final DataOutputStream dataOutputStream) throws IOException {
         dataOutputStream.writeInt(this.inventoryType);
         dataOutputStream.writeShort(this.inventory.length);
-        for (int i2 = 0; i2 < this.inventory.length; ++i2) {
-            if (this.inventory[i2] == null) {
+        for (ItemStack itemStack : this.inventory) {
+            if (itemStack == null) {
                 dataOutputStream.writeShort(-1);
+            } else {
+                dataOutputStream.writeShort((short) itemStack.itemID);
+                dataOutputStream.writeByte((byte) itemStack.stackSize);
+                dataOutputStream.writeShort((short) itemStack.itemDmg);
             }
-            else {
-                dataOutputStream.writeShort((int)(short)this.inventory[i2].itemID);
-                dataOutputStream.writeByte((int)(byte)this.inventory[i2].stackSize);
-                dataOutputStream.writeShort((int)(short)this.inventory[i2].itemDmg);
-            }
         }
     }
-    
+
     @Override
     public void processPacket(final NetHandler netHandler) {
         netHandler.handlePlayerInventory(this);
     }
-    
+
     @Override
     public int getPacketSize() {
