--- Packet22Collect.java
+++ Packet22Collect.java
@@ -1,3 +1,3 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 import java.io.DataOutputStream;
@@ -8,13 +8,13 @@
     public int collectedEntityId;
     public int collectorEntityId;
-    
+
     public Packet22Collect() {
     }
-    
+
     public Packet22Collect(final int collectedEntityID, final int collectorEntityID) {
         this.collectedEntityId = collectedEntityID;
         this.collectorEntityId = collectorEntityID;
     }
-    
+
     @Override
     public void readPacketData(final DataInputStream dataInputStream) throws IOException {
@@ -22,5 +22,5 @@
         this.collectorEntityId = dataInputStream.readInt();
     }
-    
+
     @Override
     public void writePacket(final DataOutputStream dataOutputStream) throws IOException {
@@ -28,10 +28,10 @@
         dataOutputStream.writeInt(this.collectorEntityId);
     }
-    
+
     @Override
     public void processPacket(final NetHandler netHandler) {
         netHandler.handleCollect(this);
     }
-    
+
     @Override
     public int getPacketSize() {
