--- PlayerNBTManager.java
+++ PlayerNBTManager.java
@@ -1,3 +1,3 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 import java.io.InputStream;
@@ -11,13 +11,13 @@
     public static Logger logger;
     private File playerNBT;
-    
+
     static {
         PlayerNBTManager.logger = Logger.getLogger("Minecraft");
     }
-    
+
     public PlayerNBTManager(final File playerNBT) {
         (this.playerNBT = playerNBT).mkdir();
     }
-    
+
     public void writePlayerNBT(final EntityPlayerMP entityPlayerMP) {
         try {
@@ -25,6 +25,6 @@
             entityPlayerMP.writeToNBT(nBTTagCompound2);
             final File file3 = new File(this.playerNBT, "_tmp_.dat");
-            final File file4 = new File(this.playerNBT, String.valueOf(entityPlayerMP.username) + ".dat");
-            CompressedStreamTools.writeCompressed(nBTTagCompound2, (OutputStream)new FileOutputStream(file3));
+            final File file4 = new File(this.playerNBT, entityPlayerMP.username + ".dat");
+            CompressedStreamTools.writeCompressed(nBTTagCompound2, new FileOutputStream(file3));
             if (file4.exists()) {
                 file4.delete();
@@ -33,13 +33,13 @@
         }
         catch (Exception exception5) {
-            PlayerNBTManager.logger.warning(new StringBuilder("Failed to save player data for ").append(entityPlayerMP.username).toString());
+            PlayerNBTManager.logger.warning("Failed to save player data for " + entityPlayerMP.username);
         }
     }
-    
+
     public void readPlayerNBT(final EntityPlayerMP entityPlayerMP) {
         try {
-            final File file2 = new File(this.playerNBT, String.valueOf(entityPlayerMP.username) + ".dat");
+            final File file2 = new File(this.playerNBT, entityPlayerMP.username + ".dat");
             if (file2.exists()) {
-                final NBTTagCompound nBTTagCompound3 = CompressedStreamTools.readCompressed((InputStream)new FileInputStream(file2));
+                final NBTTagCompound nBTTagCompound3 = CompressedStreamTools.readCompressed(new FileInputStream(file2));
                 if (nBTTagCompound3 != null) {
                     entityPlayerMP.readFromNBT(nBTTagCompound3);
@@ -48,5 +48,5 @@
         }
         catch (Exception exception4) {
-            PlayerNBTManager.logger.warning(new StringBuilder("Failed to load player data for ").append(entityPlayerMP.username).toString());
+            PlayerNBTManager.logger.warning("Failed to load player data for " + entityPlayerMP.username);
         }
     }
