--- ThreadLoginVerifier.java
+++ ThreadLoginVerifier.java
@@ -1,3 +1,3 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 import java.io.Reader;
@@ -9,10 +9,10 @@
     final Packet1Login loginPacket;
     final NetLoginHandler loginHandler;
-    
+
     ThreadLoginVerifier(final NetLoginHandler netLoginHandler1, final Packet1Login packet1Login2) {
         this.loginHandler = netLoginHandler1;
         this.loginPacket = packet1Login2;
     }
-    
+
     public void run() {
         try {
