--- WorldServer.java
+++ WorldServer.java
@@ -1,3 +1,3 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 import java.util.ArrayList;
@@ -12,5 +12,5 @@
     private SpawnerAnimals monsterSpawner;
     private SpawnerAnimals animalSpawner;
-    
+
     public WorldServer(final File file1, final String string2, final boolean z3) {
         super(file1, string2);
@@ -20,5 +20,5 @@
         this.monsters = z3;
     }
-    
+
     @Override
     public void tick() {
@@ -29,19 +29,18 @@
         this.animalSpawner.onUpdate(this);
     }
-    
+
     @Override
     protected IChunkProvider getChunkProvider(final File saveDir) {
         return this.chunkProviderServer = new ChunkProviderServer(this, new ChunkLoader(saveDir, true), new ChunkProviderGenerate(this, this.randomSeed));
     }
-    
+
     public List getTileEntityList(final int i1, final int i2, final int i3, final int i4, final int i5, final int i6) {
         final ArrayList arrayList7 = new ArrayList();
-        for (int i7 = 0; i7 < this.loadedTileEntityList.size(); ++i7) {
-            final TileEntity tileEntity9 = (TileEntity)this.loadedTileEntityList.get(i7);
-            if (tileEntity9.xCoord >= i1 && tileEntity9.yCoord >= i2 && tileEntity9.zCoord >= i3 && tileEntity9.xCoord < i4 && tileEntity9.yCoord < i5 && tileEntity9.zCoord < i6) {
-                arrayList7.add(tileEntity9);
+        for (TileEntity tileEntity : this.loadedTileEntityList) {
+            if (((TileEntity) tileEntity).xCoord >= i1 && ((TileEntity) tileEntity).yCoord >= i2 && ((TileEntity) tileEntity).zCoord >= i3 && ((TileEntity) tileEntity).xCoord < i4 && ((TileEntity) tileEntity).yCoord < i5 && ((TileEntity) tileEntity).zCoord < i6) {
+                arrayList7.add((TileEntity) tileEntity);
             }
         }
-        return (List)arrayList7;
+        return arrayList7;
     }
 }
