--- WorldGenReed.java
+++ WorldGenReed.java
@@ -1,3 +1,3 @@
-package net.minecraft.src;
+package net.minecraft.server;
 
 import java.util.Random;
@@ -8,10 +8,9 @@
         for (int i6 = 0; i6 < 20; ++i6) {
             final int i7 = x + rand.nextInt(4) - rand.nextInt(4);
-            final int i8 = y;
             final int i9 = z + rand.nextInt(4) - rand.nextInt(4);
             if (world.getBlockId(i7, y, i9) == 0 && (world.getBlockMaterial(i7 - 1, y - 1, i9) == Material.water || world.getBlockMaterial(i7 + 1, y - 1, i9) == Material.water || world.getBlockMaterial(i7, y - 1, i9 - 1) == Material.water || world.getBlockMaterial(i7, y - 1, i9 + 1) == Material.water)) {
                 for (int i10 = 2 + rand.nextInt(rand.nextInt(3) + 1), i11 = 0; i11 < i10; ++i11) {
-                    if (Block.reed.canBlockStay(world, i7, i8 + i11, i9)) {
-                        world.setBlock(i7, i8 + i11, i9, Block.reed.blockID);
+                    if (Block.REED.canBlockStay(world, i7, y + i11, i9)) {
+                        world.setBlock(i7, y + i11, i9, Block.REED.blockID);
                     }
                 }
